spec/android_generate_po_file_from_metadata_test.rb

```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+
+describe Fastlane::Actions::AnGeneratePoFileFromMetadataAction do
+  it 'create the .po files based on the .txt files in metadata_directory' do
+    in_tmp_dir do |dir|
+      required_keys = %w[description keywords name release_notes].freeze
+      # required_files = required_keys.map { |key| File.join(dir, "#{key}.txt") }
+
+      # For each key create a key.txt file whose content is "value key"
+      required_keys.each do |key|
+        write_to = File.join(dir, "#{key}.txt")
+        File.write(write_to, "value #{key}")
+      end
+
+      output_po_path = File.join(dir, 'PlayStoreStrings.po')
+
+      run_described_fastlane_action(
+        metadata_directory: dir,
+        release_version: '1.0'
+      )
+
+      expected = <<~PO
+        msgctxt "play_store_keywords"
```

Comment:
Note that one additional requirement that we have (see the RfC P2) is that the `.po` file also contains `# .translators: …` comments before each entry, as those comments will be picked by our translation platform (GlotPress) that will import that `.po` file, and use those comments to provide hints to translators about what each key is supposed to be about (and mention to them if some keys have maximum length limits/constraints imposed by PlayStore / AppStore)

So you will need to find a way to have those comments provided by the release-toolkit (for standard keys) — or, if you plan to also supposed custom keys as `additional_sources` for this action in the future, potentially being able to provide a comment for each custom key at call site

Reply:
I thought that `Rfc P2` meant Part 2 so it took me a while to understand :) Of course P2 means P2!
To be honest I had not noticed at all that comment in API snippet from the Rfc P2

Thanks for the feedback, I will be adding support for this next!

Reply:
> Note that one additional requirement that we have (see the RfC P2) is that the `.po` file also contains `# .translators: …` comments before each entry, as those comments will be picked by our translation platform (GlotPress) that will import that `.po` file, and use those comments to provide hints to translators about what each key is supposed to be about (and mention to them if some keys have maximum length limits/constraints imposed by PlayStore / AppStore)
> 
> So you will need to find a way to have those comments provided by the release-toolkit (for standard keys) — or, if you plan to also supposed custom keys as `additional_sources` for this action in the future, potentially being able to provide a comment for each custom key at call site

After pondering a few solutions, I think the easiest and quickest way is to add a `comments.json` in `metadata_folder`.  Something like:
`
{
  "key": "comment for key",
  "key2": "comment for key2\n2nd line comment",
}
`
I imagine the suggestions for translators are not overly complicated, a one liner could probably do. Should we need a multiline comment, we can still use `\n` in the comment. Adding comments for keys in `additional_sources` becomes easy now, it's just another entry in the `.json`.

We would have to create this file once and be happy with it. We still stay loyal to our source of truth rule.

If we ever need a more complicated comment, we could start looking at `yaml` [that handles multiline strings very well](https://yaml-multiline.info/).

Reply:
I like the `comments.json` idea. Tbh I'd directly switch to `comments.yaml` from the get go to already allow us more flexibility if we need it, and because YAML is already a superset of JSON (so we can still write that `comments.yaml` using the JSON syntax and parse it with a YAML parser and it will work too)

The only main drawback (or is it one?) of this idea though, is that this means the comments file will have to be replicated in each client repo, while in 99% of cases, we will always want the same comments (especially the part in the comment telling the translators about the character limit for a given entry) for all iOS projects, and same comments for all Android projects. So if Google and the Play Store updates their character limit one day, we'd have to update the comments in all client repos, instead of just updating the toolkit just once.

So I think a more sustainable solution would just be to store that mapping in a constant in the release-toolkit itself. We could always consider allowing the caller to override some comments via a `ConfigItem` (using the idea that is lightly mentioned in the RfC post about `other_sources` to provide additional files or override some defaults), but we'd still want to have good defaults for those comments in the first place, so that for those 99% cases the client would not have to copy/paste the comments file.

Reply:
> I like the `comments.json` idea. Tbh I'd directly switch to `comments.yaml` from the get go to already allow us more flexibility if we need it, and because YAML is already a superset of JSON (so we can still write that `comments.yaml` using the JSON syntax and parse it with a YAML parser and it will work too)

👋! Unfortunately I was already working on it early this morning before I got this message. I have just merged this [PR](https://github.com/Riziero/release-toolkit/pull/2) that's based on `comments.json` approach.

> The only main drawback (or is it one?) of this idea though, is that this means the comments file will have to be replicated in each client repo, while in 99% of cases, we will always want the same comments (especially the part in the comment telling the translators about the character limit for a given entry) for all iOS projects, and same comments for all Android projects. So if Google and the Play Store updates their character limit one day, we'd have to update the comments in all client repos, instead of just updating the toolkit just once.
>
> So I think a more sustainable solution would just be to store that mapping in a constant in the release-toolkit itself. We could always consider allowing the caller to override some comments via a `ConfigItem` (using the idea that is lightly mentioned in the RfC post about `other_sources` to provide additional files or override some defaults), but we'd still want to have good defaults for those comments in the first place, so that for those 99% cases the client would not have to copy/paste the comments file.

I 100% agre on this. Defining comments in `release-toolkit` would have been the quickest. I see how annoying it is to maintain that file in all repos.

The reason I went with a `.json/.yaml` is the usual _source of truth_ 

Mixing up the two ideas:

-  If a `comments.json` is found, use its values for comments in the po. 
- If not found, fallback to using those defined in `release-toolkit`.

How do you like this idea? Thank you :)

Reply:
The usual approach to provide potential overrides for those comments would be to just pas them as parameters to the call site (like the "API at a glance" section of the RfC suggests).

After all, that's usually what those parameters (defined via `ConfigItem` in custom fastlane actions) are used for, they provide decent default values while still allowing you to customize the values in the usually-rare cases when the default don't fit your needs.

_It'd then be up to the call site to decide how to provide those comment overrides as parameters in the (doubtful) case some client apps would need to override them (eg `additional_sources = JSON.parse('comments.json')` or something like that in the call site in the Fastfile… though at this point the more common practice would be to define it as a Ruby constant in the Fastfile directly like we do for other similar cases of shared parameter values of other actions call sites)_

So TL;DR: the usual approach is to use a `ConfigItem` with the right defaults and that's it.

Reply:
If it's so unlikely that a `comments.json` will ever be needed I do not mind  reverting my merge commit. 

If you agree I will get rid of the current  `comments.json` logic and let `toolkit-release` provide the comments.

Reply:
Works for me.